teps/0098-workflows.md

```diff
@@ -412,85 +437,416 @@ and Pipeline definitions into a higher level opinionated syntax that is aware of
+    - name: revision
+      value: "{{ revision }}"
+  pipelineSpec:
+    params:
+      - name: repo_url
+      - name: revision
+    workspaces:
+      - name: source
+    tasks:
+      - name: fetch-repository
+        taskRef:
+          name: git-clone
+        workspaces:
+          - name: output
+            workspace: source
+        params:
+          - name: url
+            value: $(params.repo_url)
+          - name: revision
+            value: $(params.revision)
+      - name: unittests
+        runAfter:
+          - fetch-repository
+        workspaces:
+          - name: source
+            workspace: source
+        taskRef:
+          name: pytest
+  workspaces:
+  - name: source
+    volumeClaimTemplate:
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+```
 
-How will UX be reviewed and by whom?
+Statuses of PipelineRuns triggered by Pipelines as Code are stored on the Repo CRD, for example:
 
-Consider including folks that also work outside the WGs or subproject.
--->
+```yaml
+apiVersion: pipelinesascode.tekton.dev/v1alpha1
+kind: Repository
+metadata:
+  name: pipelines-repo
+spec:
+  url: https://github.com/tektoncd/pipeline
+pipelinerun_status:
+- completionTime: "2022-11-17T17:56:57Z"
+  conditions:
+  - lastTransitionTime: "2022-11-17T17:56:57Z"
+    message: 'Tasks Completed: 2 (Failed: 0, Cancelled 0), Skipped: 0'
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
+  event_type: push
+  logurl: https://giphy.com/search/random-dogs
+  pipelineRunName: ci-pipelinerun-web-app-vdrh2
+  sha: 12345
+  sha_url: https://github.com/tektoncd/pipeline/commit/12345
+  startTime: "2022-11-17T17:54:54Z"
+  target_branch: main
+  title: Example commit message
+```
 
-### Performance (optional)
+### Proposed API
 
-<!--
-Consideration about performance.
-What impact does this change have on the start-up time and execution time
-of task and pipeline runs? What impact does it have on the resource footprint
-of Tekton controllers as well as task and pipeline runs?
+No changes are proposed to the installation process.
+However, Workflows will use a new "Workflow" CRD defined in the repo instead of a PipelineRun with annotations.
+There are a few differences:
 
-Consider which use cases are impacted by this change and what are their
-performance requirements.
--->
+- **Variable replacement**: The existing API uses syntax like `{{ repo_owner }}` within a PipelineRun, but this syntax does not work if defined
+in a PipelineRun yaml and applied to a cluster. Instead, variable substitutions for repo metadata are permitted only within `workflow.spec.pipelineRun`,
+and Workflows will substitute these values to create a valid PipelineRun. (This is similar to the pattern used in TriggerTemplates.)
+Variable replacement will use `$(context.foo)` syntax instead of `{{ foo }}` syntax, for consistency with existing Tekton projects.
```

Comment:
I am not sure about the argument for "not work if defined in a PipelineRun yaml and applied to a cluster.". 

The advantage of using a straight PipelineRun is that tooling have no issues reading the pac pipelinerun, the templates variable need to be defined, but the PipelineRun visualization tools still works....

I wonder if we could somehow make this inside a custom resolver instead, where we could have a PipelineRun with those `{{ variables }}` and when the resolver sees it it would replace them with values coming from TriggerTemplates variables (or other CRD who has thos values) ?